src/pages/index.js

```diff
@@ -1,94 +1,217 @@
-  handleAddCardSubmit
-);
-addCardPopup.setEventListeners();
+function renderCard(cardData) {
+  const element = createCard(cardData);
+  cardSection.addItem(element);
+}
 
-const editProfilePopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-editProfilePopup.setEventListeners();
+/* ------------------------ USER INFO AND INITAL CARS ----------------------- */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "60774818-2b72-4a3b-8d83-5b2abf9c8a5e",
+    "Content-Type": "application/json",
+  },
+});
 
-const imagePreview = new PopupWithImage("#popup-modal");
-imagePreview.setEventListeners();
+api
+  .loadPageContent()
+  .then(([cards, userData]) => {
+    newUserInfo.setUserAvatar(userData.avatar);
+    newUserInfo.setUserInfo({
+      name: userData["name"],
+      about: userData["about"],
+      id: userData["_id"],
+    });
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-const renderCard = (cardData) => {
-  const createCard = new Card(cardData, "#card-template", handleImageClick);
-  cardSection.addItem(createCard.getView());
-};
+let cardSection;
 
-const cardSection = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
+const newUserInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-cardSection.renderItems();
 
-const profileEditFormValidator = new FormValidator(config, profileEditForm);
-profileEditFormValidator.enableValidation();
-
-const addCardFormValidator = new FormValidator(config, addCardForm);
-addCardFormValidator.enableValidation();
-
-function handleImageClick(title, link) {
-  imagePreview.open(title, link);
-}
+/* ------------------------------ ADD NEW CARD ------------------------------ */
+const addCardPopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+addCardPopup.setEventListeners();
 
 function handleAddCardSubmit(data) {
-  renderCard(data);
-  addCardPopup.close();
+  addCardPopup.setLoading(true);
+  api
+    .addCard(data)
+    .then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => addCardPopup.setLoading(false, "Create"));
 }
 
+addCardOpenButton.addEventListener("click", () => {
+  formValidators["edit-profile-form"].resetValidation();
+  addCardPopup.open();
+});
+
+/* ------------------------------ EDIT PROFILE ------------------------------ */
+const editProfilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+editProfilePopup.setEventListeners();
+
 function handleProfileEditSubmit(data) {
-  newUserInfo.setUserInfo(data);
-  editProfilePopup.close();
+  editProfilePopup.setLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```